### DISCUSSION TITLE
Copy directory from repository to local folder

### DISCUSSION THREAD

**Discussion by HeinziTuberkel** [community/community#23772] [New to GitHub]
<p>Hi, </p>
<p>I’m a single developer using git to document my project changes and versions.</p>
<p>I’m pretty new to git and I still have problems understanding some of the concepts behind it.</p>
<p>In the past decades I used to just make dater, commented backups of my projects as a “repository”.</p>
<p>I have some projects in a git repository with several commits and three branches.</p>
<p>Now I want to merge some files of my current project with the ones in a different branch of the repository. I just can’t find a way to save a copy of the other revision to a different directory.</p>
<p>My Project has the branches “D7” and “D10”. It contains a Directory “Customers”.</p>
<p>My active project directory is in sync with the branch “D7”.</p>
<p>Now I want to merge some content from files in directory “Customers” of branch “D10” to my working directory. </p>
<p>To do that I want to extract the complete directory “Customers” to a temporary folder so I can compare the two folders and merge file contents with Beyond Compare.</p>
<p>Final question:</p>
<p>How can I copy files from a certain revision of the repository to some local directory without destroying the contents of my working directory?</p>

**Comment by yamiacat** [comment]
<p>Hi heinzituberkel! :wave: Are you using Git on the command line?</p>
<p>When you <code>git checkout</code> to a different branch, the entire contents of the repository change to the version of the files on that branch. You can, however, checkout just specific files from other branches, using:</p>
<p><code>git checkout branch_name file_path/file_name.ext</code></p>
<p>This will bring a copy of the specified file from the specified branch into your current branch. Assuming the files in different branches have different names, this would work for you. If you’re on branch <code>D7</code> currently you can use:</p>
<p><code>git checkout D10 Customers/file_name.ext</code></p>
<p>If the files on each branch have the same names, though, you won’t be able to do this, as it will just swap out the <code>D7</code> version for the <code>D10</code> version of the file with that name.</p>
<p>In that case, you could save a copy of the file to a temporary directory that is <em>outside</em> the repository, switch to the other branch, and repeat the process. So, if you’re in the Customers folder:</p>
<p><span class="hashtag">#copy</span> the file to a folder outside the repo and rename it<br>
<code>cp file_name.ext ../../temp/filenameD7version.ext</code></p>
<p><span class="hashtag">#switch</span> branch<br>
<code>git checkout d10</code></p>
<p><span class="hashtag">#copy</span> the file to the same temp folder outside the repo and rename it<br>
<code>cp file_name.ext ../../temp/filenameD10version.ext</code></p>
<p>Then you can use Beyond Compare on the files.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="heinzituberkel" data-post="1" data-topic="3217">
<div class="title">
<div class="quote-controls"></div>
 heinzituberkel:</div>
<blockquote>
<p>How can I copy files from a certain revision of the repository to some local directory without destroying the contents of my working directory?</p>
</blockquote>
</aside>
<p>The <a href="https://git-scm.com/docs/git-worktree"><code>git worktree</code> command</a> can do what you want. E.g. if your current directory is your existing repository, and you want the branch <code>some-branch</code> in a directory named <code>repo-some-branch</code> next to your primary directory:</p>
<pre><code class="lang-bash">git worktree add ../repo-some-branch some-branch
</code></pre>
<p>The new directory is bound to your repository for the git data, but the checkout is independent, so you can copy files and commit on either side as usual. You could also use this to work on two branches separately.</p>
<p>Note however that if you want to fully merge the branches (rather than just importing a few files from one to the other, whether using the <code>checkout</code> method described by <a class="mention" href="/yamiacat">@yamiacat</a> or worktrees) a <code>git merge</code> may be a better solution.</p>

**Comment by HeinziTuberkel** [comment]
<p>Thanks a lot! The “worktree add” was just what I was looking for.</p>
<p>BTW: in the past weeks I did something similar to yamicat’s suggestion: I copied the complete working directory to a different location, checked out the other branch and did the compare.</p>
<p>Problem solved :smile:</p>
